src/main/java/frc/lib/logfields/logfields/LogField.java

```diff
@@ -0,0 +1,49 @@
+package frc.lib.logfields.logfields;
+
+import java.util.function.Supplier;
+
+import org.littletonrobotics.junction.LogTable;
+import org.littletonrobotics.junction.inputs.LoggableInputs;
+
+public class LogField<T> implements Supplier<T>, LoggableInputs {
+    private final Supplier<T> valueSupplier;
+    private final String name;
+    private final LogTableGetFunc<T> logTableGetFunc;
+    private final LogTablePutFunc<T> logTablePutFunc;
+    private T value;
+
+    public interface LogTablePutFunc<T> {
+        void putInTable(LogTable table, String name, T value);
+    }
+
+    public interface LogTableGetFunc<T> {
+        T getFromTable(LogTable table, String name, T defaultValue);
+    }
+
+    public LogField(
+            String name,
+            Supplier<T> valueSupplier,
+            LogTableGetFunc<T> logTableGetFunc,
+            LogTablePutFunc<T> logTablePutFunc) {
+        this.name = name;
```

Comment:
A space between the constructor and the this.---- values

Reply:
I actually tried it when I wrote the code, and it didn't improve clarity and made it look a bit out of place.